### PR TITLE
Path determination improvements

### PR DESCRIPTION
**Pull request by freyacodes**
* Refactor path determination
* Added configuration of doses path via `SKRIVE_DOSES_PATH`
* Removed homebrew path

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 # Skrive
 Secure and sleek dosage logging for the terminal.
 
-# Usage
+## Usage
 `skrive [path to doses.dat]`
+
+Skrive looks for a doses file in this order:
```
Comment (VanuPhantom) [inline README.md]: The original text might be interpreted to imply that *Skrive* will move on to the next option if opening a file at the given path fails. This is not the case.

```suggestion
Skrive selects a path for the doses file in the following order:
```

File: README.md [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+
+1. At the path of the program argument, as shown above
```
Comment (VanuPhantom) [inline README.md]:
```suggestion
1. The path provided as the first argument, as shown above
```

File: README.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
...
+2. At the path given by environment variable `SKRIVE_DOSES_PATH`, if set
```
Comment (VanuPhantom) [inline README.md]:
```suggestion
2. The path given by the environment variable `SKRIVE_DOSES_PATH`, if set
```

File: README.md [hunk 4]
```diff
@@ -1,5 +1,13 @@
...
+3. At `~/doses.dat`
```
Comment (VanuPhantom) [inline README.md]:
```suggestion
3. `~/doses.dat` (A file called `doses.dat` in your home directory.)
```

File: logic/logic.go [hunk 5]
```diff
@@ -14,10 +15,35 @@ type Dose struct {
...
+		dosageFilePath = value
+		return nil
+	}
+
+	if dirname, err := os.UserHomeDir(); err != nil {
+		return homePathError{}
+	} else {
+		dosageFilePath = path.Join(dirname, "doses.dat")
+		return nil
+	}
```
Comment (VanuPhantom) [inline logic/logic.go:36]: I wonder whether it would be more appropriate for the happy path to be the first arm of the if/else statement.

**Review by VanuPhantom — CHANGES_REQUESTED** [review]
This is an excellent addition to *Skrive*. I do think some slight alterations would be appropriate regarding the documentation.

I especially appreciate that you've removed the preliminary attempts at supporting [Homebrew](https://brew.sh/). Those should be reintroduced when we bring in full Homebrew support.

File: README.md (hunk 4 above)
Comment (freyacodes) [reply README.md]: The sentence in parenthesis is redundant to the sentence below it

File: logic/logic.go (hunk 5 above)
Comment (freyacodes) [reply logic/logic.go:36]: I like to take error handling off my mind before handling the happy path, e.g. with an early return

File: README.md (hunk 4 above)
Comment (VanuPhantom) [reply README.md]: The line does not pertain to the creation of a file.

*Skrive* should ideally support Windows. Users on that platform will most likely not be familiar with the shorthand `~/doses.dat`. Thusly, I've added a clarification.

We might need to be even more clear, though.

```suggestion
3. The path to a file called `doses.dat` in your home directory
    - `~/doses.dat` on Unix-like platforms
    - `C:\Users\<user>\doses.dat` on Windows
```

File: logic/logic.go (hunk 5 above)
Comment (VanuPhantom) [reply logic/logic.go:36]: That's a good point. Your approach is better at conveying the presence of a control flow interruption. Let's stick with it.

**Review by VanuPhantom — APPROVED** [review]
Perfect!

**Pull request merged**